fl4health/feature_alignment/tab_features_preprocessor.py

```diff
@@ -0,0 +1,147 @@
+
+    def get_default_binary_pipeline(self) -> Pipeline:
+        return Pipeline(steps=[("imputer", SimpleImputer(strategy="most_frequent")), ("encoder", OrdinalEncoder())])
+
+    def get_one_hot_pipeline(self, categories: MetaData) -> Pipeline:
+        return Pipeline(steps=[("encoder", OneHotEncoder(handle_unknown="ignore", categories=[categories]))])
+
+    def get_ordinal_pipeline(self, categories: MetaData) -> Pipeline:
+        return Pipeline(
+            steps=[
+                (
+                    "encoder",
+                    OrdinalEncoder(
+                        unknown_value=len(categories) + 1,
+                        handle_unknown="use_encoded_value",
+                        categories=[categories],
+                    ),
+                )
+            ]
+        )
+
+    def get_default_string_pipeline(self, vocabulary: MetaData) -> Pipeline:
+        return Pipeline(steps=[("vectorizer", TextColumnTransformer(TfidfVectorizer(vocabulary=vocabulary)))])
+
+    def initialize_default_pipelines(
+        self, tabular_features: List[TabularFeature], feature_to_transformer: Dict[str, Pipeline], one_hot: bool
+    ) -> None:
+        """
+        Initialize a default Pipeline for every data column in tabular_features.
+
+        Parameters
+        ----------
+            tabular_features (List[TabularFeature]): list of tabular
+            features in the data columns.
+        """
+        for tab_feature in tabular_features:
+            feature_type = tab_feature.get_feature_type()
+            feature_name = tab_feature.get_feature_name()
+            if feature_type == TabularType.NUMERIC:
+                feature_pipeline = self.get_default_numeric_pipeline()
+            elif feature_type == TabularType.BINARY:
+                feature_pipeline = self.get_default_binary_pipeline()
+            elif feature_type == TabularType.ORDINAL:
+                feature_categories = tab_feature.get_metadata()
+                if one_hot:
+                    feature_pipeline = self.get_one_hot_pipeline(feature_categories)
+                else:
+                    feature_pipeline = self.get_ordinal_pipeline(feature_categories)
+            else:
+                vocabulary = tab_feature.get_metadata()
+                feature_pipeline = self.get_default_string_pipeline(vocabulary)
+            feature_to_transformer[feature_name] = feature_pipeline
+
+    def return_column_transformer(self, pipelines: Dict[str, Pipeline]) -> ColumnTransformer:
+        transformers = [
+            (f"{feature_name}_pipeline", pipelines[feature_name], [feature_name])
+            for feature_name in sorted(pipelines.keys())
+        ]
+        return ColumnTransformer(
+            transformers=transformers,
+            remainder="drop",
+        )
+
+    def set_feature_pipeline(self, feature_name: str, pipeline: Pipeline, target: bool) -> None:
```

Comment:
In the example that you implemented, maybe we can show how this function is used somewhere? That is, we set a custom pipeline for the feature. It can be the same as the default, but demonstrating the use would be nice.

Reply:
Yes good idea!